front/src/Components/Video/VideoMediaBox.svelte

```diff
@@ -1,46 +1,72 @@
 <script lang="ts">
-    import type {VideoPeer} from "../../WebRtc/VideoPeer";
+    import type { VideoPeer } from "../../WebRtc/VideoPeer";
     import SoundMeterWidget from "../SoundMeterWidget.svelte";
     import microphoneCloseImg from "../images/microphone-close.svg";
     import reportImg from "./images/report.svg";
     import blockSignImg from "./images/blockSign.svg";
-    import {videoFocusStore} from "../../Stores/VideoFocusStore";
-    import {showReportScreenStore} from "../../Stores/ShowReportScreenStore";
-    import {getColorByString, srcObject} from "./utils";
+    import { videoFocusStore } from "../../Stores/VideoFocusStore";
+    import { showReportScreenStore } from "../../Stores/ShowReportScreenStore";
+    import { userWokaPictureStore } from "../../Stores/UserWokaPictureStore";
+    import { getColorByString, srcObject } from "./utils";
+    import { onDestroy } from "svelte";
 
     export let peer: VideoPeer;
     let streamStore = peer.streamStore;
     let name = peer.userName;
     let statusStore = peer.statusStore;
     let constraintStore = peer.constraintsStore;
 
+    let userWokaPictureSrc: string | undefined = undefined;
+
+    const unsubscribeFromUserWokaPictureStore = userWokaPictureStore.subscribe((playersAvatars) => {
+        userWokaPictureSrc = playersAvatars.get(peer.userId);
+        console.log(userWokaPictureSrc);
```

Comment:
Oopsie, missed that one!

As for the refactor idea, I really like it :) Gonna proceed with the changes.

For the failing tests of renderTexture not having width to work with - I wonder if it may be due to the headless browser?